packages/kbn-i18n/GUIDELINE.md

```diff
@@ -0,0 +1,330 @@
+}
+```
+
+Mostly such placeholders have meaningful name according to the сontent.
+
+### Pluralization
+
+I18n engine supports proper plural forms. It uses the [ICU Message syntax](http://userguide.icu-project.org/formatparse/messages) to define a message that has a plural label and works for all [CLDR languages](http://cldr.unicode.org/) which have pluralization rules defined. The numeric input is mapped to a plural category, some subset of "zero", "one", "two", "few", "many", and "other" depending on the locale and the type of plural.
+
+For example:
+
+```js
+{
+  'kbn.management.createIndexPattern.step.status.successLabel.strongIndicesLabel': '{indicesLength, plural, one {# index} other {# indices}}'
+}
+```
+
+In case when `indicesLength` has value 1, the result string will be "`1 index`". In case when `indicesLength` has value 2 and more, the result string - "`2 indices`".
+
+## Best practices
+
+
+### Naming convention
+
+The message ids chosen for message keys should always be descriptive of the string, and its role in the interface (button label, title, etc.). Think of them as long variable names. When you have to change a message id, adding a progressive number to the existing key should always be used as a last resort.
+
+- Message id should start with namespace (`kbn`, `common.ui` and etc.). 
+
+    For example:
+
+  ```js
+  'kbn.management.createIndexPattern.stepTime.options.patternHeader'
+  'common.ui.indexPattern.warningLabel'
+  ```
+
+- Use camelCase for naming segments, comprising several words.
+
+- Each message id should end with a type. For example:
+
+  ```js
+  'kbn.management.editIndexPattern.createIndexButton'
+  'kbn.management.editIndexPattern.mappingConflictHeader'
+  'kbn.management.editIndexPattern.mappingConflictLabel'
+  'kbn.management.editIndexPattern.fields.filterAria'
+  'kbn.management.editIndexPattern.fields.filterPlaceholder'
+  'kbn.management.editIndexPattern.refreshTooltip'
+  'kbn.management.editIndexPattern.fields.allTypesDropDown'
+  'kbn.management.createIndexPattern.includeSystemIndicesToggleSwitch'
+  'kbn.management.editIndexPattern.wrongTypeErrorMessage'
+  'kbn.management.editIndexPattern.scripted.table.nameDescription'
+  ```
+
+- For complex messagges, that is divided into several parts, use the folllowing approach:
+  - the main message id should have the type on the penultimate position, thereby identifying a divided phrase, and the last segment should end with `Detail`,
+  - the inner message id should have the type on the penultimate position and the name of the variable from the placeholder in the main message as the last segment that ends with its own type.
+
+  For example before the translation there was a message:
+
+  ```js
+  <strong>Success!</strong>
+  Your index pattern matches <strong>{exactMatchedIndices.length} {exactMatchedIndices.length > 1 ? 'indices' : 'index'}</strong>.
```

Comment:
this ternary is wrong for english. plural is used for everything but the number 1. so it should be `{exactMatchedIndices.length === 1 ? 'index' : 'indices'}`